test/gtest/ucs/test_sock.cc

```diff
@@ -53,6 +69,50 @@ class test_socket : public ucs::test {
             EXPECT_EQ(exp_str, test_str);
         }
     }
+
+    void verify_subnet_match(sa_family_t af, const char *ip_addr1,
+                             const char *ip_addr2,
+                             unsigned longest_common_prefix) const
+    {
+        struct sockaddr_storage storage1, storage2;
+        struct sockaddr *saddr1;
+        struct sockaddr *saddr2;
+        int matched;
+
+        saddr1 = (struct sockaddr*)&storage1;
+        saddr2 = (struct sockaddr*)&storage2;
+        fill_sockaddr(saddr1, af, ip_addr1, 1234);
+        fill_sockaddr(saddr2, af, ip_addr2, 1234);
+
+        size_t addr_size_bits = get_addr_size(saddr1) * CHAR_BIT;
+
+        for (size_t prefix = 0; prefix <= addr_size_bits; ++prefix) {
+            ASSERT_UCS_OK(ucs_sockaddr_is_same_subnet(saddr1, saddr2, prefix,
+                                                      &matched));
+            EXPECT_EQ(prefix <= longest_common_prefix, matched)
+                    << "failed to match subnet: ip1=" << ip_addr1
+                    << " ip2=" << ip_addr2 << " prefix=" << prefix
+                    << " max_common=" << longest_common_prefix
+                    << " (matched=" << matched << ")";
```

Comment:
why needed? prefix is defined by addr_size_bits
can iterate from 0 to std::min(addr_size_bits, longest_common_prefix)

Reply:
the idea is to check all possible sizes and decide whether they should be matched or not.
`longest_common_prefix` is used as the threshold between matched/unmatched subnet.